### PR TITLE
feat(pit-crew): add Pit Crew action with Radar mode (#375)

### LINKED ISSUE #375: Add Pit Engineer action: voice-driven race engineer
## What would you like?

Add a new **Pit Engineer** action that plays voice callouts for pit lane, race, and spotter events based on live iRacing telemetry. One button toggles the engineer on/off; each sub-feature can be enabled or disabled independently from the Property Inspector.

### Sub-features

**Pit lane**

- Pit approach alert
- Service reminders (including auto-fuel heads-up)
- Stall-departure pit-limiter reminder
- Pit exit callout
- Pit-limiter warnings (no-limiter, limiter dropped, speeding in lane)

**Race**

- Flag alerts (yellow / blue / green / black / red / white / checkered)
- Incident alerts (off-track, spin, contact) with material sampling
- Toggle confirmations (DRS, Push-to-Pass, fuel, fast repair, tires, windshield, compound change / stay)
- Overtake callouts and racing tips

**Fuel (WIP, disabled by default)**

- Laps-remaining / pit-window / refuel warnings
- Stint-opener fuel report
- Mid-stint on-plan / high-consumption checkpoints
- Save-fuel lift-and-coast coaching

**Spotter**

- Directional callouts (left / right / both / two-left / two-right)
- Self-scheduling tick loop with per-state intervals
- Suppressed on pit road and in Lone Qualify

**Other**

- Welcome message with driver-name slot-fill and racing-tip picker
- Opt-in engine-startup orange-snake light sweep animation

### Property Inspector

- Master toggle + per-sub-feature checkboxes (nested accordions)
- Independent Engineer / Spotter volume sliders with Test buttons
- Audio output device dropdown (System Default + enumerated devices)
- Driver-name autocomplete
- Standard title / color / border / graphic / common-settings sections

## Use case

Current iRaceDeck actions are visual-only. A voice-engineer channel adds a race-engineer experience during driving — hands-on-wheel information about the race state, pit sequencing, and situational awareness — without adding more keys to look at or using a separate tool like Crew Chief. Because iRaceDeck already has deep telemetry integration, the engineer can react to pit-service toggles, tire compound choices, and session state directly.

## Area

Action (new)

## Alternatives considered

- Leaving this to external tools (Crew Chief, SimHub): works, but requires a separate process and duplicates config that iRaceDeck already has.
- Adding audio one sub-feature at a time through smaller actions: more granular but fragments the PI across many buttons and duplicates the audio-engine plumbing.

## Additional context

- Implementation on \`feature/pit-engineer\` (2 commits ahead of \`master\`).
- New shared package \`@iracedeck/audio-assets\` holds MP3 clips; both plugins copy them into \`assets/audio/\` at build time.
- New \`deck-core\` audio service (mixer + bus layer) and \`iracing-native\` miniaudio engine (vendored \`miniaudio.h\`) provide the runtime.
- Fuel sub-feature ships disabled (WIP); can be opted into per-user.
- Code review will want to look closely at the \`pit-engineer.ts\` size (~3.5k LOC, many module globals) — a modularization plan already exists at \`docs/plans/2026-04-19-pit-engineer-modularization-plan.md\` and will be tracked as a separate follow-up if/when triggered.

### PR DESCRIPTION
**Pull request by niklam**
## Related Issue

Closes #375

## What changed?

Adds the **Pit Crew** action (formerly *Pit Engineer*) — a voice-driven race engineer backed by live iRacing telemetry.

- **Radar mode** ships in GA: directional spotter callouts (left / right / two-left / two-right), suppressed on pit road and in Lone Qualify, with independent volume and test playback
- **Race Engineer mode** is shipped hidden until voice scenarios are finalized (tracked as a follow-up)
- New packages:
  - `@iracedeck/audio-native` — miniaudio-based native engine (vendored `miniaudio.h`)
  - `@iracedeck/audio-service` — mixer, bus layer, base-path resolution
  - `@iracedeck/audio-assets` — MP3 library under `pit-crew/` namespace (flags, incidents, pitlane, radar, toggle, tips, …)
  - `@iracedeck/audio-scenarios` — scenario engine (ported from the original pit-engineer action)
  - `@iracedeck/event-bus` + `@iracedeck/sim-events-iracing` — telemetry→event translator (flags, fuel, incidents, lifecycle, limiter, overtakes, pit lane, radar, toggles)
- PI additions:
  - `ird-audio-device-select` — audio output device picker (System Default + enumerated)
  - `ird-audio-test` — inline test-playback button
  - `ird-range-input` — 0 is now a valid value (enables radar volume = 0)
- Output Device is persisted by stable `ma_device_id` rather than enumeration index
- Global-settings cache is now updated synchronously on local writes to avoid stale reads

## How to test

- Install the plugin and add a **Pit Crew** button
- In the Property Inspector:
  - Pick an audio output device; use **Test** to confirm playback
  - Enable Radar, set volume, use Test to verify the bus
  - Set radar volume to 0 and confirm the slider accepts it
- In iRacing, with traffic nearby:
  - Verify directional callouts (left / right / both / two-left / two-right)
  - Drive onto pit road → callouts must stop
  - Start a Lone Qualify session → callouts must not fire
- Toggle the radar off and back on → key stays green only while enabled; toggling off must disable (no re-enable loop)
- Build both plugins and verify the action registers on Stream Deck and Mirabox

## Checklist

- [x] Linked to an approved issue
- [x] New code has unit tests
- [x] Changes registered in all applicable plugins (Stream Deck, Mirabox)
- [x] No unrelated changes included

## Summary by CodeRabbit

* **New Features**
  * Added "Pit Crew" Stream Deck action with Radar and Radar Volume modes, audio test and persistent output-device selection.
  * Multi‑channel audio engine, scenario-driven Pit Crew voice/alert catalog (flags, fuel, pit, incidents, overtakes, tips) and runtime radar previews.
  * Simulator semantic event bus to coordinate telemetry-driven audio and features.
  * New UI components for audio device selection and audio test controls.

* **Documentation**
  * Extensive audio architecture, modularization, incident-alerts, Pit Crew and integration docs; README updated to reflect new actions/modes.

### CODE REVIEW

File: .claude/rules/stream-deck-actions.md [hunk 1]
```diff
@@ -179,6 +179,26 @@ Always include both scripts in PI HTML files:
...
+
+**`ird-audio-device-select`** - Plugin-global audio output device dropdown. Binds to a global setting storing the selected device index, and populates its options from a second global setting that holds the device-list JSON (maintained by the plugin at runtime).
+```html
+<sdpi-item label="Output Device">
+  <ird-audio-device-select setting="audioOutputDevice" devices="_audioDeviceList" default-label="System Default"></ird-audio-device-select>
+</sdpi-item>
+```
+- `setting` - Global setting key holding the selected device index (default: `audioOutputDevice`)
+- `devices` - Global setting key holding the device-list JSON (default: `_audioDeviceList`). Payload: `[{ "index": number, "name": string, "isDefault"?: boolean }, …]`
+- `default-label` - Label for the `-1` (System Default) option (default: `System Default`)
```
[the remark this answers is not in the document]
Comment (niklam) [reply .claude/rules/stream-deck-actions.md]: Fixed in 0776dec9 — updated the `setting` / `devices` / `default-label` bullets to describe the id-based contract (payload now `{ id, name, isDefault }`, empty string = System Default) and added a note about the persist-on-fallback behavior so future PI authors don't reintroduce the index-based assumption.

File: packages/audio-native/scripts/build.mjs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+if (isWindows) {
+  console.log("Building native addon (node-gyp rebuild)...");
+  try {
+    execSync("node-gyp rebuild", { stdio: "inherit" });
+  } catch {
+    if (existsSync(addonPath)) {
+      console.warn("node-gyp rebuild failed (file may be locked by a running process). Using existing native addon.");
+    } else {
+      throw new Error("node-gyp rebuild failed and no existing addon found. Close any processes using the addon and retry.");
+    }
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/audio-native/scripts/build.mjs]: Fixed in de305dc2 — catch captures the error, inspects `err.code` for `EBUSY`/`EPERM` and the message for `in use by another process` / `being used by another process` / `permission denied`, and only falls back to the existing `.node` when one of those matches. Everything else rethrows, so compiler errors and missing toolchain failures aren't papered over anymore.

File: packages/pi-components/src/components/audio-device-select.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+    // If the persisted device id is no longer in the current option list
+    // (e.g. unplugged headset, or a legacy numeric-index value from
+    // pre-#427 builds), fall back to System Default so the dropdown
+    // reflects what miniaudio will actually use. Native
+    // `<select>.value = "<unknown>"` silently clears the selection —
+    // surface that as the System Default value explicitly.
+    const exists = Array.from(this.select.options).some((opt) => opt.value === this.savedValue);
+    this.select.value = exists ? this.savedValue : SYSTEM_DEFAULT_VALUE;
+  }
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/pi-components/src/components/audio-device-select.ts:202]: Fixed in 9befc7a7. One twist: the setting callback routinely fires before the devices callback on PI open, so naively persisting on every `!exists` apply would overwrite the user's real selection during that race. Added a `devicesLoaded` flag — `applySavedValue` only writes the fallback once the device-list payload has actually arrived. Covered by two new tests: one verifies `save("")` fires when the saved id is genuinely missing, another verifies no redundant persist happens when the saved value is already System Default.

File: packages/sim-events-iracing/src/diff/incidents.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  state.materialHistory = state.materialHistory.filter((s) => now - s.t <= MATERIAL_WINDOW_MS);
+
+  // First tick — seed without firing.
+  if (state.lastIncidentCount < 0) {
+    state.lastIncidentCount = incidentCount;
+
+    return;
+  }
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/sim-events-iracing/src/diff/incidents.ts]: Fixed in 9e591d4b. The first-tick branch now seeds both the incident counter AND the off-track fields from the current snapshot, then returns early. If the snapshot already shows `surface === TrkLoc.OffTrack` in an eligible zone, `offTrackStartedAt` / `offTrackPending` / `materialHistory` get seeded to match so the normal transition logic doesn't synthesize `offTrack.started` on the next tick. New test `does not synthesize offTrack.started when the first tick starts off-track (mid-excursion reconnect)` covers the reconnect-while-off-track path.

File: packages/sim-events-iracing/src/diff/incidents.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+  const delta = incidentCount - state.lastIncidentCount;
+  state.lastIncidentCount = incidentCount;
+
+  if (delta > 0) {
+    emit({ event: "incident.occurred", data: {} });
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/sim-events-iracing/src/diff/incidents.ts]: Fixed in 9e591d4b. `incident.occurred` now carries `{ delta: number }` (catalog updated to match). The existing `pit-crew.incident-alerts` scenario only filters on event name so it keeps working; the TODO in that scenario file about branching on severity can now ride on `delta` directly. Test `forwards the incident delta in the event payload` asserts `delta=1` for a 1x track-limits nudge vs `delta=4` for a 4x spin/contact.

File: packages/sim-events-iracing/src/diff/lifecycle.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+  // ── Engine startup (RPM 0 → >threshold) ────────────────────────────────
+  const rpm = typeof telemetry.RPM === "number" ? telemetry.RPM : 0;
+  const engineRunning = rpm > ENGINE_STARTUP_RPM_THRESHOLD;
+
+  if (!state.lastEngineRunning && engineRunning) {
+    emit({ event: "engine.startup", data: {} });
+  }
+
+  state.lastEngineRunning = engineRunning;
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/sim-events-iracing/src/diff/lifecycle.ts]: Fixed in 9e591d4b. Added a `lifecycleInitialized` flag to translator state; the first tick seeds `firstOnTrackFired`, `lastSessionNum`, `lastEngineRunning`, and `lastLap` from the current snapshot and returns. Reconnecting with the engine already running no longer fires a bogus `engine.startup` — new test `does not synthesize engine.startup when the first tick already has RPM > threshold` covers that. Same init also suppresses a replay `driver.firstOnTrack` on a reconnect that isn't actually the driver's first on-track moment, which matters because the welcome scenario is gated purely on that event.

File: packages/sim-events-iracing/src/diff/lifecycle.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+  // ── Lap started ────────────────────────────────────────────────────────
+  const lap = typeof telemetry.Lap === "number" ? telemetry.Lap : -1;
+
+  if (lap > 0 && state.lastLap > 0 && lap !== state.lastLap) {
+    emit({ event: "lap.started", data: { lap } });
+  }
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/sim-events-iracing/src/diff/lifecycle.ts]: Fixed in 9e591d4b. Changed the guard to `lap > state.lastLap` so a session reset like Lap 12 → 1 stops being treated as a lap-start. Test `does not synthesize lap.started when Lap resets to a lower number (session flip)` covers the practice → race transition.

File: packages/sim-events-iracing/src/diff/pit-lane.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    state.approachExitingSuppressed = true;
+  } else if (!isApproaching && !onPitRoad) {
+    // Left approach zone cleanly — clear suppression.
+    state.approachExitingSuppressed = false;
+  }
+
+  if (isApproaching && !state.approachAlertFired && !isExitingPits) {
+    state.approachAlertFired = true;
+    emit({ event: "pitLane.approaching", data: {} });
+  } else if (isOnTrack && state.approachAlertFired) {
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/sim-events-iracing/src/diff/pit-lane.ts:69]: Fixed in 9e591d4b. Added a `pitLaneInitialized` flag; first tick seeds `lastOnPitRoad`, `lastInPitStall`, `approachAlertFired`, and `approachExitingSuppressed` from the current snapshot and returns. When the first tick is already on pit road or in the approach zone, `approachExitingSuppressed` is seeded `true` (matching where the existing exit-suppression logic would have landed after observing the entry). Two new tests cover the mid-stall and mid-approach reconnect paths.

File: packages/sim-events-iracing/src/translator.ts [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+
+// ── Internals ──────────────────────────────────────────────────────────────
+
+function handleDisconnect(self: TranslatorInstance): void {
+  // Fresh state so reconnect seeds cleanly.
+  self.state = createInitialState();
+  self.latestTelemetry = null;
+  self.pitSpeedLimitMps = 0;
+  self.pitSpeedLimitKey = "";
+}
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/sim-events-iracing/src/translator.ts:155]: Fixed in 2b45b28d. `handleDisconnect` now publishes a `radar.changed { from, to: "clear" }` envelope before wiping state when `state.radarState !== "clear"`. The radar engine already handles `to: "clear"` by stopping the tick loop, so the teardown drops in cleanly without needing a new event type. Guarded on `latestTelemetry !== null` to cover a connect-immediately-disconnect edge case. Radar is the only active-state subsystem today; others that land later should plug in here the same way. Two new tests cover active-radar disconnect (emits the teardown) and clear-radar disconnect (stays silent).

File: packages/audio-native/scripts/build.mjs [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    const isLockOrPermissionError =
+      err?.code === "EBUSY" ||
+      err?.code === "EPERM" ||
+      /in use by another process|being used by another process|permission denied/i.test(message);
+
+    if (isLockOrPermissionError && existsSync(addonPath)) {
+      console.warn("node-gyp rebuild failed (file may be locked by a running process). Using existing native addon.");
+    } else {
+      throw error;
+    }
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/audio-native/scripts/build.mjs:31]: Fixed in c86940dd — applied the same selective catch (EBUSY / EPERM / "in use by another process" / "permission denied") to `iracing-native/scripts/build.mjs` so both native build scripts surface real failures instead of silently reusing a stale `.node`. Added a "keep in sync" comment on both sides to keep future touches aligned.

File: packages/sim-events-iracing/src/translator.test.ts [hunk 11]
```diff
@@ -0,0 +1,769 @@
...
+
+  controller.__tick = (telemetry, isConnected = true) => {
+    callback?.(telemetry, isConnected);
+  };
+  controller.__setSessionInfo = (info) => {
+    sessionInfo = info;
+  };
+
+  return controller;
+}
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/sim-events-iracing/src/translator.test.ts:75]: Fixed in 05175ef0 — the mock's `subscribe` now invokes `cb(null, false)` immediately after storing the callback, matching the real `SDKController.subscribe` which fires `callback(this.sdk.getTelemetry(), this.isConnected)` right after adding the subscriber. `(null, false)` reflects the real controller's initial state (offline, no telemetry yet). All existing tests still pass — the translator's `handleTick` bails on null telemetry and `handleDisconnect` is a no-op on fresh state, so the immediate call is benign for both paths.

**Comment by niklam** [comment]
Nitpick from review [4169794682](https://github.com/niklam/iracedeck/pull/430#pullrequestreview-4169794682) addressed in 3cfc3acc — added an unconditional `vi.useRealTimers()` to the suite `afterEach` so the overtake test's fake timers can't leak into later tests if anything between `useFakeTimers()` and the inline `useRealTimers()` throws.

**Comment by niklam** [comment]
Nitpick from [review 4169811293](https://github.com/niklam/iracedeck/pull/430#pullrequestreview-4169811293) — no change. Grepped `packages/sim-events-iracing/src` (non-test) for `createScope`: zero usages. The translator never chains on the return value, so the current `vi.fn()` stub returning `undefined` is safe. The nitpick's own caveat ("If the translator never chains on `createScope` results, this is fine as-is") applies.
